### PR TITLE
Alerting: Add SaveAndApply methods to the forked Alertmanager (remote secondary)

### PR DESCRIPTION
**Pull request by santihernandezc**
This PR adds `SaveAndApplyConfig()` and `SaveAndApplyDefaultConfig()` to the forked Alertmanager struct in remote secondary mode.

### CODE REVIEW

File: pkg/services/ngalert/remote/forked_alertmanager_test.go [hunk 1]
```diff
@@ -21,6 +21,32 @@ func TestForkedAlertmanager_ModeRemoteSecondary(t *testing.T) {
 	ctx := context.Background()
 	expErr := errors.New("test error")
 
+	t.Run("SaveAndApplyConfig", func(tt *testing.T) {
+		// SaveAndApplyConfig should only be called on the remote Alertmanager.
+		// State and configuration are updated on an interval.
+		internal, _, forked := genTestAlertmanagers(tt, modeRemoteSecondary)
+		internal.EXPECT().SaveAndApplyConfig(ctx, mock.Anything).Return(nil).Once()
+		require.NoError(tt, forked.SaveAndApplyConfig(ctx, &apimodels.PostableUserConfig{}))
```
Comment (gotjosh) [inline pkg/services/ngalert/remote/forked_alertmanager_test.go:29]: shouldn't we also assert that remote was not called?

File: pkg/services/ngalert/remote/remote_secondary_forked_alertmanager.go [hunk 2]
```diff
@@ -25,11 +25,15 @@ func (fam *RemoteSecondaryForkedAlertmanager) ApplyConfig(ctx context.Context, c
 }
 
 func (fam *RemoteSecondaryForkedAlertmanager) SaveAndApplyConfig(ctx context.Context, config *apimodels.PostableUserConfig) error {
```
Comment (gotjosh) [inline pkg/services/ngalert/remote/remote_secondary_forked_alertmanager.go:27]:
```suggestion
// SaveAndApplyConfig when running in remote secondary, is only called in the internal Alertmanager.
func (fam *RemoteSecondaryForkedAlertmanager) SaveAndApplyConfig(ctx context.Context, config *apimodels.PostableUserConfig) error {
```

File: pkg/services/ngalert/remote/remote_secondary_forked_alertmanager.go [hunk 3]
```diff
@@ -25,11 +25,15 @@ func (fam *RemoteSecondaryForkedAlertmanager) ApplyConfig(ctx context.Context, c
...
-	return nil
+	// We only need to call this method on the internal Alertmanager.
+	// Configuration is synced on an interval.
+	return fam.internal.SaveAndApplyConfig(ctx, config)
 }
 
 func (fam *RemoteSecondaryForkedAlertmanager) SaveAndApplyDefaultConfig(ctx context.Context) error {
```
Comment (gotjosh) [inline pkg/services/ngalert/remote/remote_secondary_forked_alertmanager.go:33]:
```suggestion
// SaveAndApplyDefaultConfig when running in remote secondary, is only called in the internal Alertmanager.
func (fam *RemoteSecondaryForkedAlertmanager) SaveAndApplyDefaultConfig(ctx context.Context) error {
```

**Review by gotjosh — APPROVED** [review]
LGTM and approving to unblock you but please take a look at my test comment.

File: pkg/services/ngalert/remote/forked_alertmanager_test.go (hunk 1 above)
Comment (santihernandezc) [reply pkg/services/ngalert/remote/forked_alertmanager_test.go:29]: If you make an unexpected call, the test fails.

**Pull request merged**